make test before patch:
```
bash-4.2# NO_INTERACTION='1'   REPORT_EXIT_STATUS='1'   TEST_PHP_ARGS='-d zend_extension=/usr/lib64/php/modules/opcache.so' make test
/bin/sh /home/builder/rpmbuild/BUILD/shadow-0.8.1/libtool --mode=compile cc  -I. -I/home/builder/rpmbuild/BUILD/shadow-0.8.1 -DPHP_ATOM_INC -I/home/builder/rpmbuild/BUILD/shadow-0.8.1/include -I/home/builder/rpmbuild/BUILD/shadow-0.8.1/main -I/home/builder/rpmbuild/BUILD/shadow-0.8.1 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic   -c /home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c -o shadow.lo
libtool: compile:  cc -I. -I/home/builder/rpmbuild/BUILD/shadow-0.8.1 -DPHP_ATOM_INC -I/home/builder/rpmbuild/BUILD/shadow-0.8.1/include -I/home/builder/rpmbuild/BUILD/shadow-0.8.1/main -I/home/builder/rpmbuild/BUILD/shadow-0.8.1 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -c /home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c  -fPIC -DPIC -o .libs/shadow.o
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c: In function 'clean_cache_dir':
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c:730:8: warning: unused variable 'c' [-Wunused-variable]
   char c;
        ^
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c: In function 'shadow_stream_opener':
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c:783:5: warning: implicit declaration of function 'php_stream_parse_fopen_modes'; did you mean 'php_stream_free_enclosed'? [-Wimplicit-function-declaration]
  if(php_stream_parse_fopen_modes(mode, &flags) == SUCCESS) {
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
     php_stream_free_enclosed
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c: In function 'shadow_is_writable':
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c:1292:19: warning: unused variable 'new_name' [-Wunused-variable]
  zval *old_name, *new_name;
                   ^~~~~~~~
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c:1292:8: warning: unused variable 'old_name' [-Wunused-variable]
  zval *old_name, *new_name;
        ^~~~~~~~
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c:1291:9: warning: unused variable 'name' [-Wunused-variable]
  zval **name;
         ^~~~
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c: In function 'shadow_glob':
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c:1335:7: warning: unused variable 'num' [-Wunused-variable]
  long num;
       ^~~
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c:1327:9: warning: unused variable 'name' [-Wunused-variable]
  zval **name;
         ^~~~
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c: In function 'shadow_generic_override':
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c:1486:19: warning: unused variable 'new_name' [-Wunused-variable]
  zval *old_name, *new_name;
                   ^~~~~~~~
/home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow.c:1486:8: warning: unused variable 'old_name' [-Wunused-variable]
  zval *old_name, *new_name;
        ^~~~~~~~
/bin/sh /home/builder/rpmbuild/BUILD/shadow-0.8.1/libtool --mode=compile cc  -I. -I/home/builder/rpmbuild/BUILD/shadow-0.8.1 -DPHP_ATOM_INC -I/home/builder/rpmbuild/BUILD/shadow-0.8.1/include -I/home/builder/rpmbuild/BUILD/shadow-0.8.1/main -I/home/builder/rpmbuild/BUILD/shadow-0.8.1 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic   -c /home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow_cache.c -o shadow_cache.lo
libtool: compile:  cc -I. -I/home/builder/rpmbuild/BUILD/shadow-0.8.1 -DPHP_ATOM_INC -I/home/builder/rpmbuild/BUILD/shadow-0.8.1/include -I/home/builder/rpmbuild/BUILD/shadow-0.8.1/main -I/home/builder/rpmbuild/BUILD/shadow-0.8.1 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -c /home/builder/rpmbuild/BUILD/shadow-0.8.1/shadow_cache.c  -fPIC -DPIC -o .libs/shadow_cache.o
/bin/sh /home/builder/rpmbuild/BUILD/shadow-0.8.1/libtool --mode=link cc -DPHP_ATOM_INC -I/home/builder/rpmbuild/BUILD/shadow-0.8.1/include -I/home/builder/rpmbuild/BUILD/shadow-0.8.1/main -I/home/builder/rpmbuild/BUILD/shadow-0.8.1 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic  -Wl,-z,relro -o shadow.la -export-dynamic -avoid-version -prefer-pic -module -rpath /home/builder/rpmbuild/BUILD/shadow-0.8.1/modules  shadow.lo shadow_cache.lo
libtool: link: cc -shared  -fPIC -DPIC  .libs/shadow.o .libs/shadow_cache.o    -O2 -m64 -mtune=generic -Wl,-z -Wl,relro   -Wl,-soname -Wl,shadow.so -o .libs/shadow.so
libtool: link: ( cd ".libs" && rm -f "shadow.la" && ln -s "../shadow.la" "shadow.la" )
/bin/sh /home/builder/rpmbuild/BUILD/shadow-0.8.1/libtool --mode=install cp ./shadow.la /home/builder/rpmbuild/BUILD/shadow-0.8.1/modules
libtool: install: cp ./.libs/shadow.so /home/builder/rpmbuild/BUILD/shadow-0.8.1/modules/shadow.so
libtool: install: cp ./.libs/shadow.lai /home/builder/rpmbuild/BUILD/shadow-0.8.1/modules/shadow.la
libtool: finish: PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/sbin" ldconfig -n /home/builder/rpmbuild/BUILD/shadow-0.8.1/modules
----------------------------------------------------------------------
Libraries have been installed in:
   /home/builder/rpmbuild/BUILD/shadow-0.8.1/modules

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------

Build complete.
Don't forget to run 'make test'.


=====================================================================
PHP         : /usr/bin/php
PHP_SAPI    : cli
PHP_VERSION : 7.1.26
ZEND_VERSION: 3.1.0
PHP_OS      : Linux - Linux 9d2e25b923ef 4.9.125-linuxkit #1 SMP Fri Sep 7 08:20:28 UTC 2018 x86_64
INI actual  : /home/builder/rpmbuild/BUILD/shadow-0.8.1/tmp-php.ini
More .INIs  :
CWD         : /home/builder/rpmbuild/BUILD/shadow-0.8.1
Extra dirs  :
VALGRIND    : Not used
=====================================================================
TIME START 2019-03-13 01:06:20
=====================================================================
SKIP Check chgrp() [tests/chgrp.phpt]
SKIP Check chmod() [tests/chmod.phpt]
SKIP Check chown() [tests/chown.phpt]
PASS Check listing directory content [tests/dir.phpt]
PASS Check working with directories [tests/dirs.phpt]
PASS Check file_exists [tests/exists.phpt]
PASS Check fread with specific length [tests/fread.phpt]
PASS Check shadow_get_config [tests/getconfig.phpt]
PASS Check globs [tests/glob.phpt]
PASS Check is_writable on instance [tests/is_writable.phpt]
PASS Check directory listing for main dir [tests/maindir.phpt]
PASS Check creating directories [tests/mkdir.phpt]
PASS Check creating directories, full paths [tests/mkdir2.phpt]
PASS Check creating directories when shadow is not enabled [tests/mkdir3.phpt]
SKIP Check overrides [tests/override.phpt]
SKIP Check overrides [tests/override_class.phpt]
PASS Check reading from files [tests/read.phpt]
PASS Check reading from instance-only dirs [tests/read_custom.phpt]
PASS Check listing directory content [tests/readdir.phpt]
PASS Check realpath+open (issue 2) [tests/realpath-open.phpt]
PASS Check renaming [tests/rename.phpt]
PASS Check require [tests/require.phpt]
PASS Check require_once [tests/require_once.phpt]
PASS Check require_once with other requires [tests/require_once2.phpt]
PASS Check require_once with other requires [tests/require_once3.phpt]
PASS Check require_once with opcache [tests/require_once_opcache.phpt]
PASS Check shadow settings change in script [tests/shadow_settings.phpt]
PASS Check touching files with stream [tests/stream.phpt]
PASS Check touching files [tests/touch.phpt]
PASS Check unlinking files [tests/unlink.phpt]
PASS Check unlinking files and then reading [tests/unlink2.phpt]
PASS Check write+rename [tests/write-rename.phpt]
PASS Check writing to files [tests/write.phpt]
PASS Check writing to files [tests/write_custom.phpt]
PASS Check writing to files in new dir [tests/write_new.phpt]
PASS Check writing to files in new dir, full paths [tests/write_new2.phpt]
SKIP Check writing to files in new dir with dir mask [tests/write_new_mask.phpt]
=====================================================================
TIME END 2019-03-13 01:06:22

=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   16
---------------------------------------------------------------------

Number of tests :   37                31
Tests skipped   :    6 ( 16.2%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    0 (  0.0%) (  0.0%)
Expected fail   :    0 (  0.0%) (  0.0%)
Tests passed    :   31 ( 83.8%) (100.0%)
---------------------------------------------------------------------
Time taken      :    2 seconds
=====================================================================
bash-4.2#
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,7 +3,7 @@ sudo: yes
 dist: trusty
 
 env:
-  - PHPVER="7.1.7"
+  - PHPVER="7.1.27"
 
 # This replaces the travis supplied phpbuild which defaults to ZTS enabled
 # builds.  Since we are replacing using the same config, we don't have to add

--- a/shadow.c
+++ b/shadow.c
@@ -216,7 +216,6 @@ static void shadow_override_function(char *fname, size_t fname_len, int argno, i
 
     zend_string_release(fname_zs);
     zend_string_release(fname_full);
-    pefree(override, 1);
 }
 
 /* {{{ PHP_MINIT_FUNCTION

